c2cgeoform/templates/grid.jinja2

```diff
@@ -1,70 +1,133 @@
+
+      {# see: https://github.com/wenzhixin/bootstrap-table/tree/develop/src/locale #}
+      {% set bootstrap_table_locales = {
+          'de': 'de-DE',
+          'en': 'en-US',
+          'fr': 'fr-FR',
+      } %}
+      locale: '{{ bootstrap_table_locales[request.locale_name] }}',
+
+      uniqueId: '_id_',
+      columns: [
+        {
+          field: 'actions',
+          formatter: function(value, row, index, field) {
+            var content = '\
+              <div class="btn-group">\
+                <button type="button" class="btn btn-xs btn-default dropdown-toggle" data-toggle="dropdown" data-target="item-' + row._id_ + '-actions" aria-haspopup="true" aria-expanded="false">\
+                  <span class="caret"></span>\
+                </button>\
+                <ul id="item-' + row._id_ + '-actions" class="dropdown-menu">';
+
+            row.actions.forEach(function(action) {
+              if (action.confirmation || action.method) {
+                content += '\
+                  <li>\
+                    <a class="c2cgeoform-item-action c2cgeoform-action-ajax ' + action.name + ' ' + action.css_class + '"\
+                       href="#"\
+                       data-url="' + action.url + '"\
+                       data-method="' + action.method + '"\
+                       data-confirmation="' + action.confirmation + '">\
+                       <span class="' + action.icon + '"></span>\
+                       ' + action.label + '\
+                    </a>\
+                  </li>';
+              } else {
+                content += '\
+                  <li>\
+                    <a class="c2cgeoform-item-action ' + action.name + ' ' + action.css_class + '"\
+                       href="' + action.url + '">\
+                       <span class="' + action.icon + '"></span>\
+                       ' + action.label + '\
+                    </a>\
+                  </li>';
+              }
+            });
+
+            content += '\
+                </ul>\
+              </div>';
+
+            return content
+            return value;
           }
-      }
-    }).on("loaded.rs.jquery.bootgrid", function(e, appendedRows) {
+        }
+      ]
+    });
+
+    $('#grid').on('page-change.bs.table', function(e, number, size) {
+      localStorage.setItem('pageSize', size);
+    });
 
-      $("a.btn-delete", this).on('click', function() {
-        if (window.confirm('{{_("Are your sure ?")}}')) {
+    $('#grid').on('load-success.bs.table', function (e) {
+
+      $(this).find('a.c2cgeoform-action-ajax').on('click', function(e) {
+        var execute = function() {
           $.ajax({
-              url: $(this).attr("data-url"),
-              type: 'DELETE',
-              success: function(result) {
-                  location.reload();
+            url: $(this).data('url'),
+            type: $(this).data('method'),
+            success: function(data) {
+              if (this.type == 'DELETE') {
```

Comment:
look strange... maybe define a behaviour (and test for it rather than testing on method) or redefine redirection rules, not redirecting to index on delete... may need extra thinking.

Reply:
delete should be the only one case where we want to redirect to index view. In case of grid, we do not want to reload the page but only the json content.